### DISCUSSION TITLE
How to customize Form.List component logic?

### DISCUSSION THREAD

**Discussion by Lucjus** [ant-design/ant-design#35046] [Q&A] [answered]
Hello everyone, 

I am building dynamic form with the use of antd form.list. I need to customize some logic here. In addition to the methods: `add, remove, move`  I would like to add a new one called `duplicate` which will duplicate existing form item with values. I have no idea where to even start. 
Is there a way to do such thing?

https://ant.design/components/form/

**Comment by huntdream** [comment, marked as answer]
You can pass `defaultValue` to `add` 
<img width="1176" alt="image" src="https://user-images.githubusercontent.com/7903755/163668548-47e511d4-9fce-4598-b580-de26b92dcde5.png">

**Reply by Lucjus** [reply]
This will do, thanks! :)
